src/lib/disk_cache/lmdb/disk_cache.ml

```diff
@@ -16,29 +16,63 @@ module Make (Data : Binable.S) = struct
 
   module Rw = Read_write (F)
 
-  type t = { env : Rw.t; db : Rw.holder; counter : int ref }
+  type t =
+    { env : Rw.t
+    ; db : Rw.holder
+    ; counter : int ref
+    ; logger : Logger.t
+    ; garbage : int Hash_set.t
+          (** A list of ids that are no longer reachable from OCaml's side *)
+    }
+
+  (** How big can the above hashset be before we do a cleanup *)
+  let garbage_size_limit = 512
 
   let initialize path ~logger =
     Async.Deferred.Result.map (Disk_cache_utils.initialize_dir path ~logger)
       ~f:(fun path ->
         let env, db = Rw.create path in
-        { env; db; counter = ref 0 } )
+        { env
+        ; db
+        ; counter = ref 0
+        ; logger
+        ; garbage = Hash_set.create (module Int)
+        } )
 
   type id = { idx : int }
 
-  let get ({ env; db; _ } : t) ({ idx } : id) : Data.t =
+  let get ({ env; db; logger; _ } : t) ({ idx } : id) : Data.t =
+    [%log debug] "Getting data at %d in LMDB cache" idx
```

Comment:
Is it a useful log?

Reply:
Well, it could be for us to understanding delaying when loading a lot of caches.

Reply:
Should it not be spam? Just in case we hit many times?

Reply:
I mark it spam in follow up PRs